components/pkg-export-helm/src/maintainer.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2018 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::str::FromStr;
+use std::string::ToString;
+use std::result;
+use clap::ArgMatches;
+use url::Url;
+
+use export_docker::Result;
+
+use error::Error;
+
+#[derive(Clone, Debug)]
+pub struct Maintainer {
+    name: String,
+    email: Option<String>,
+    url: Option<String>,
+}
+
+impl Maintainer {
+    pub fn from_args(matches: &ArgMatches) -> Result<Vec<Self>> {
+        let mut maintainers = Vec::new();
+
+        if let Some(args) = matches.values_of("MAINTAINER") {
+            for arg in args {
+                let m = arg.parse::<Self>()?;
+
+                maintainers.push(m);
+            }
+        };
+
+        Ok(maintainers)
+    }
+}
+
+impl FromStr for Maintainer {
+    type Err = Error;
+
+    /// Creates a `Maintainer` struct from a string representation, which must be of the format
+    /// `NAME[,EMAIL[,URL]]`.
+    ///
+    /// # Errors
+    ///
+    /// * `maintainer_str` is not of the format `NAME[,EMAIL[,URL]`
+    /// * An invalid URL is specified
+    fn from_str(maintainer_str: &str) -> result::Result<Self, Self::Err> {
+        let values: Vec<&str> = maintainer_str.split(',').collect();
+        if values.len() < 1 || values.len() > 3 {
+            return Err(Error::InvalidMaintainer(maintainer_str.to_owned()));
+        }
+
+        let name = values[0].to_string();
+        // FIXME: Check validity of email address
+        let email = values.get(1).map(|&s| s.to_owned());
+        let url = values.get(2).map(|&s| s.to_owned());
+        if let Some(ref u) = url {
+            Url::parse(&u).map_err(|e| {
+                Error::InvalidUrl(u.to_owned(), format!("{}", e))
+            })?;
+        };
+
+        Ok(Maintainer { name, email, url })
+    }
+}
+
+impl ToString for Maintainer {
+    /// Convert into a string. The returned string can be directly put in a Chart.yaml file under
+    /// the `maintainers` section.
+    fn to_string(&self) -> String {
+        let mut s = format!("  - name: {}", self.name);
+
+        if let Some(ref e) = self.email {
+            s += &format!("\n    email: {}", e);
+        };
+        if let Some(ref u) = self.url {
+            s += &format!("\n    url: {}", u);
+        };
```

Comment:
This embeds YAML formatting in the code; it would be better to handle this in the chart template itself.

Reply:
I was thinking of doing the same but since I don't know how I can access different fields/variables for each maintainer, it'll have to be a separate template file.

Reply:
I think you can just convert them to JSON objects and iterate through them in the template, if it doesn't allow you to directly access members of Rust structs.

Reply:
Oh I didn't know you can access JSON object's members from the template. I'll look into that.